homeassistant/components/sonos/switch.py

```diff
@@ -408,7 +395,6 @@ def async_migrate_speech_enhancement_entity_unique_id(
         entry
         for entry in registry_entries
         if entry.domain == Platform.SWITCH
-        and entry.original_icon == FEATURE_ICONS[ATTR_SPEECH_ENHANCEMENT]
```

Comment:
Removing this might get all switch entities for this device, since this seems to be the only condition to only match the (old) "speech enhancement" entity.

I think the condition should be kept to check for `mdi:ear-hearing`. The icon should still be in the entity registry (as it would have been used in the previous HA version).

(IMO, this is a weird way to match entities for a migration though)